buffer/src/main/java/io/netty/buffer/ReadOnlyByteBuf.java

```diff
@@ -40,7 +40,7 @@
     private final ByteBuf buffer;
 
     public ReadOnlyByteBuf(ByteBuf buffer) {
-        super(buffer.maxCapacity());
+        super(buffer.capacity()); // ReadOnly buffers are not allowed to grow capacity.
```

Comment:
I'm happy to pull these ReadOnly buffer changes into a another PR if necessary but wanted to include them here for discussion. If these ReadOnly buffers reflect their writable bytes accurately we won't have to have a special case for them in BTMD.

Note this is still inconsistent with `ReadOnlyByteBufferBuf` which uses `buffer.remaining()`, but it seems more correct than what is there now because in general because `ReadOnlyByteBuf`'s capacity cannot be adjusted and no bytes can be written to it.

Reply:
see https://github.com/netty/netty/pull/9878

Reply:
IMHO it would be better to discuss in separate PR ... I guess there's two parts - whether these changes are safe/appropriate to make to `ReadOnlyByteBuf`, and if so whether it's safe to rely on that behaviour in BTMD without checking `.isReadOnly()` (e.g. what about some "other" read only `ByteBuf` impl with prior `writableBytes()` semantics).

Interestingly I noticed what appears to be a bug related to this - there is no `isReadOnly()` check in `BTMD#discardSomeReadBytes()`. `ReadOnlyByteBuf` does not override `discardSomeReadBytes()` (only `discardReadBytes()`), which also seems like a bug, since the results would be meaningless/unpredicable.

Reply:
+1 please remove from here

Reply:
removed from this PR ... discussion over to other PR.

Reply:
@Scottmitch yes I think we need the isReadOnly check... great catch

Reply:
it was there before this PR ... what I noticed is it seems unnecessary with the updated conditionals. however we can discuss this potential optimization as part of https://github.com/netty/netty/pull/9878 as a followup to this PR.